docs/source/tutorial/introduction.rst

```diff
@@ -0,0 +1,439 @@
+########################
+Introduction to retworkx
+########################
+
+This guide serves as an introduction to working with retworkx. If you're a
+current or past `NetworkX <https:://networkx.org>`__ user who is looking at
+using retworkx as a replacement for NetworkX, you can also refer to
+:ref:`networkx` for a detailed comparison.
+
+Creating a Graph
+================
+
+To create an empty undirected graph :math:`G` with no nodes and no edges you
+can run the following code:
+
+.. jupyter-execute::
+
+    import retworkx as rx
+    G = rx.PyGraph()
+
+A :class:`~retworkx.PyGraph` is comprised of nodes (vertices)
+and unordered pairings of nodes (called edges, links, etc). Both nodes and
+edges in retworkx have an assigned data payload (also referred to as a weight
+in the API and documentation) which can be any Python object (e.g. a numeric
+value, a string, an image, an XML object, another graph, a custom node object,
+etc.). Nodes and edges are uniquely identified by an integer ``index``
+which is stable for the lifetime of that node or edge in the graph. These
+indices are not guaranteed to be contiguous as removed nodes or
+edges will leave holes in the sequence of assigned indices, and an index
+can be reused after a removal.
```

Comment:
```suggestion
edges will leave holes in the sequence of assigned indices, as other indices
are not affected by removals.
```
I think "index can be reused after a removal" gives the sense that if you remove index 1, we might assign 1 to another node later. So I tried rephrasing it

Reply:
Actually, that's exactly my intent here. If you do:

```
import retworkx
graph = retworkx.add_nodes_from(range(3))
graph.remove_node(1)
graph.add_node("NEWNODE")
graph[1] == "NEWNODE"
```

petgraph is keeping a freed node and edge list internally to avoid reallocating the internal vecs when there are holes.

Reply:
I forgot about that petgraph internal, well let's leave it as is